handlers.js

```diff
@@ -318,14 +508,330 @@
+                                                                this.setHeader("WWW-Authenticate", "Basic")
+                                                                this.write(data, 401)
+                                                                this.finish()
+                                                                this.useDefaultMime = true
+                                                            }.bind(this)
+                                                            reader.readAsText(filee)
+                                                        }.bind(this))
+                                                    } else {
+                                                        this.write('Path of 401 html was not found - 401 path is set to: '+this.app.opts.optCustom401location, 500)
+                                                        this.finish()
+                                                    }})
+                                                } else {
+                                                    this.useDefaultMime = false
+                                                    this.setHeader('content-type','text/html; charset=utf-8')
+                                                    this.setHeader("WWW-Authenticate", "Basic")
+                                                    this.write("<h1>401 - Unauthorized</h1>", 401)
+                                                    this.finish()
+                                                    this.useDefaultMime = true
+                                                    return
+                                                }
+                                            }
+                                            if (validAuth) {
+                                                htaccessCheck2.bind(this)()
+                                            }
+                                        } else {
+                                            htaccessCheck2.bind(this)()
+                                        }
+
+                            }
+                            var filerequest = this.request.origpath
+                            
+                            if (this.app.opts.optExcludeDotHtml) {
+                                this.fs.getByPath(this.request.path+'.html', (file) => {
+                                if (! file.error) {
+                                    if (this.request.origpath.endsWith("/")) {
+                                        var filerequested = filerequest.split('/').pop();
+                                        if (filerequested == 'index.html' ||
+                                            filerequested == 'index.htm' ||
+                                            filerequested == 'index' ||
+                                            filerequested == 'index.xhtm' ||
+                                            filerequested == 'index.xhtml' ||
```

Comment:
this would be much cleaner to make a helper util function, instead of duplicating it in several places

Reply:
Thats a good Idea. I may do that but its not likely.